minigalaxy/ui/library.py

```diff
@@ -89,23 +88,18 @@ def __create_gametiles(self) -> None:
         games_with_removed_tiles = []
         for child in self.flowbox.get_children():
             tile = child.get_children()[0]
-            if tile.current_state == tile.state.INSTALLED:
-                if not tile.game.image_url:
-                    games_with_removed_tiles.append(copy.deepcopy(tile.game))
-                    self.flowbox.remove(tile)
-                    continue
-            if tile.game in self.games:
+            if tile.current_state == tile.state.INSTALLED and not tile.game.image_url:
+                games_with_removed_tiles.append(copy.deepcopy(tile.game))
+            elif tile.game in self.games:
                 games_with_tiles.append(tile.game)
 
         for game in self.games:
-            if game in games_with_tiles:
-                continue
             if game in games_with_removed_tiles:
                 for tile_game in games_with_removed_tiles:
                     if game == tile_game:
                         game.install_dir = tile_game.install_dir
-                        break
```

Comment:
Does this break not do anything?

Reply:
Well, I am not sure at the moment, but I don't think there should be more than one hit between self.games and games_with_removed_tiles per ttile. If such a situation will be the case, maybe we should diagnose why is it so instead of hiding it with break statement after first hit.

Reply:
I think this break was there because it speeds up the code execution. You don't have to go through the full list if you already found the game.